File: 007 10001st prime/main.cpp
#include<iostream>
#include<cmath>


bool isCoPrime(int num, int* primes, int current) {
	for(int i=0; i<current; i++) {
		if(sqrt(num) < primes[i]) {
			return true;
		}
		if(num%primes[i] == 0) {
			return false;
		}
	}
	return true;
}

int getNextCoPrime(int* primes, int i) {
	int prime = primes[i-1]+1;
	while(!isCoPrime(prime, primes, i)) {
		prime++;
	}
	return prime;
}

int main() {
	int primes[10001] = {2};
	for(int i=1; i<10001; i++) {
		primes[i] = getNextCoPrime(primes, i);
	}
	std::cout<<primes[10000]<<std::endl;
}

File: 007 10001st prime/main.py
#!/usr/bin/python3.5
import math

primes 		= [2]

def isRelativelyPrime(num, primes):
	for x in primes:
		if(x > math.sqrt(num)):
			return True
		if(not num%x):
			return False
	return True

def getNextCoPrime(primes):
	prime = primes[-1]+1
	while(not isRelativelyPrime(prime, primes)):
		prime += 1
	return prime

for i in range(1, 10001):
	primes.append(getNextCoPrime(primes))

print("10001st prime is: " + str(primes[-1]))
